apps/ftu/js/wifi.js

```diff
@@ -105,13 +104,14 @@ var WifiManager = {
   },
 
   getNetwork: function wm_gn(ssid) {
-    var network;
-    for (var i = 0; i < this.networks.length; i++) {
-      if (this.networks[i].ssid == ssid) {
-        network = this.networks[i];
-        break;
+    var network = null;
+    this.getNetworks(function received(list) {
+      for (var i = 0; i < list.length; i++) {
+        if (list[i].ssid === ssid) {
+          network = list[i];
```

Comment:
Do you need to break here? The old implementation would stop at the first match. Yours will continue.